overlay.nix

```diff
@@ -1,4 +1,4 @@
 final: prev: {
-  inherit (final.callPackage ./builder { }) buildGoApplication mkGoEnv;
-  gomod2nix = final.callPackage ./default.nix { };
+  inherit (final.darwin.apple_sdk_11_0.callPackage ./builder { }) buildGoApplication mkGoEnv;
+  gomod2nix = final.darwin.apple_sdk_11_0.callPackage ./default.nix { };
```

Comment:
https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/platform-notes.chapter.md
it's the same as normal `callPackage` on linux.